File: py/tests/fixtures/sample_project/src/myapp/services.py
from myapp.models import User


def create_user(name: str, email: str) -> User:
    user = User(name=name, email=email)
    user.save()
    return user


def process(data: str, unused_param: int) -> str:
    """unused_param is never read — caught by pyfindunused."""
    return data.upper()

File: plugins/bonsai/ts/tests/fixtures/sample_ts_project/src/services.ts
import { User } from "./models";

export function createUser(name: string, email: string): User {
  const user = new User(name, email);
  user.save(); // call on a User instance — tsfindrefs for User.save should find this
  return user;
}

export function processData(data: string, unusedParam: number): string {
  return data.toUpperCase();
}
